src/lib.rs

```diff
@@ -6,73 +6,66 @@
 //! ```
 //! use uchardet::detect_encoding_name;
 //!
-//! assert_eq!(Ok("ISO-8859-1".to_string()),
-//!            detect_encoding_name(&[0x46, 0x72, 0x61, 0x6e, 0xe7, 0x6f, 0x69, 0x73, 0xe9]));
+//! assert_eq!("ISO-8859-1".to_string(),
+//!            detect_encoding_name(&[0x46, 0x72, 0x61, 0x6e, 0xe7, 0x6f,
+//!                0x69, 0x73, 0xe9]).unwrap());
 //! ```
 //!
 //! For more information, see [this project on
 //! GitHub](https://github.com/emk/rust-uchardet).
 
+// Increase the compiler's recursion limit for the `error_chain` crate.
+#![recursion_limit = "1024"]
 #![deny(missing_docs)]
 
+#[macro_use]
+extern crate error_chain;
 extern crate libc;
 extern crate uchardet_sys as ffi;
 
 use libc::size_t;
-use std::error::Error;
-use std::fmt;
-use std::result::Result;
 use std::ffi::CStr;
 use std::str::from_utf8;
 
-/// An error occurred while trying to detect the character encoding.
-#[derive(Debug, PartialEq)]
-pub struct EncodingDetectorError {
-    message: String
-}
-
-impl Error for EncodingDetectorError {
-    fn description(&self) -> &str { "encoding detector error" }
-    fn cause(&self) -> Option<&Error> { None }
-}
+use errors::*;
```

Comment:
Try `pub use` instead, so that users can access all the types used in our API signatures. Older Rust will fail to compile without this, and callers can't use `links` to import our error chain into theirs.